Decisions/scs-XXXX-v1-requirements-for-sso-identity-federation.md

```diff
@@ -0,0 +1,187 @@
+---
+title: Requirements for SSO identity federation
+type: Decision Record
+status: Draft
+track: IAM
+---
+
+# Introduction
+
+Our assumption is that there are use cases, where CSPs would like to be able to
+let customers access their SCS based services by identifying themselves with
+credentials that are stored and managed external to SCS.
+
+This is based on the observation that prospective customers of a SCS based CSP
+sometimes already come equipped with an IAM solution of their choice, either on
+premises or e.g. as an external 3rd party cloud service. To ease onboarding of
+customer employees (or e.g. customer contracted 3rd party admin staff) as SCS
+users, it would be good to be able to consume these external identities in SCS.
+
+For customers this avoids the neccessity to explicitly maintain an additional
+dedicated account in SCS and this also reduces what SCS needs to do with
+respect to handling storage and lifecycle maintenance for user accounts.
+
+To put it in other workds, in SCS we would like to be able to delegate
+authentication to external identity providers and map those users to roles in
+SCS that can be used for authorization decisions when users access SCS services.
+
+# Motivation for this document
+
+SCS has multiple service layers, like IaaS and CaaS, both of which running their
+own technological stack with specific internal models of accounts and
+authorization. OTOH there are also specific services like the "Status Page"
+application that require consumption of identites.
+
+One thing these services have in common, is that they are able
+to use SSO protocols like OAuth 2.0 or OpenID Connect (OIDC) on top of it to
+delegate authentication. They are service providers (SAML terminology) and can
+be relying parties (OIDC terminology) of a protocol compliant identity provider
+(IdP).
+
+So the idea is, to run an SSO IdP as part of SCS to provide a dedicated point
+of entry for identites, which the SCS service layers can use as a common
+interface to consume external user identities.
```

Comment:
Does this mean that SCS does not offer to maintain (persisted) identities itself?
Or that there are multiple IdP's? One for SCS-local users and one for federated ones?

Reply:
This paragraph only talks about the topic of external identities (the document is about federation after all). But you are right, there probably is also a need for SCS-local identities. I've only mentioned it below `ephemeral workload identities`. Maybe we should also mention this here. I'm unsure, because I don't have a clear picture for this. We could theoretically, as you mention, have one IdP for federation and use a separate IdP for storage of actual SCS-local identities. As a physicist I like the symmetry. But I think that's overkill technically. I'd rather store them either inside of the IdP or in an attached LDAP compliant IAM backend.

Reply:
+1

I think that it would be beneficial to explicitly state that maintenance of SCS-local identities is also not decided yet. Otherwise a reader might think a) that maintaining identities **within** SCS is done already and b) that the whole identity federation feature is an opt-in feature, as it is kind-of suggested above, by saying that only a subset of all customers will profit from it:

> This is based on the observation that prospective customers of a SCS based CSP
> **sometimes** already come equipped with an IAM solution of their choice

On-topic: I like the perspective of having a single IdP handling all identities - no matter where they come from (user identities primarily managed by the IdP itself; workload identities; user identities from 3rd party IdP's; etc.) as hinted here:

> So the idea is, to run a SSO IdP as part of SCS to provide a dedicated point
> of entry for identites,

Reply:
Good point, timely following up to today's discussion. Added in latest commit